### PR TITLE
BD-2014 add add bundled transactions to kotlin SDK

### PR DESCRIPTION
**Pull request by edrotthoff**
add add bundled transactions to kotlin SDK

### CODE REVIEW

File: fiosdk/src/main/java/fiofoundation/io/fiosdk/FIOSDK.kt [hunk 1]
```diff
@@ -3054,7 +3084,134 @@ class FIOSDK(private var privateKey: String, var publicKey: String,var technolog
...
+        {
+            val wfa = if(technologyPartnerId.isEmpty()) this.technologyPartnerId else technologyPartnerId
+
+            val validator = validateAddBundledTransactions(fioAddress,bundleSets,wfa)
+
+            if(!validator.isValid)
+                throw FIOError(validator.errorMessage!!)
+            else
+            {
+                val transferFioAddress = AddBundledTransactionsAction(
```
Comment (ericbutz) [inline fiosdk/src/main/java/fiofoundation/io/fiosdk/FIOSDK.kt]: transferFioAddress?

File: fiosdk/src/main/java/fiofoundation/io/fiosdk/FIOSDK.kt [hunk 2]
```diff
@@ -3054,7 +3084,134 @@ class FIOSDK(private var privateKey: String, var publicKey: String,var technolog
...
+                        fioAddress,
+                        bundleSets,
+                        maxFee,
+                        wfa,
+                        this.publicKey
+                )
+
+                val actionList = ArrayList<AddBundledTransactionsAction>()
+                actionList.add(transferFioAddress)
```
Comment (ericbutz) [inline fiosdk/src/main/java/fiofoundation/io/fiosdk/FIOSDK.kt]: should rename

File: fiosdk/src/main/java/fiofoundation/io/fiosdk/FIOSDK.kt [hunk 3]
```diff
@@ -3885,4 +4042,16 @@ class FIOSDK(private var privateKey: String, var publicKey: String,var technolog
...
+    {
+        var isValid = bundleSets > 0
+
+        isValid = isValid && fioAddress.isFioAddress()
+
+        if(technologyPartnerId.isNotEmpty())
+            isValid = isValid && technologyPartnerId.isFioAddress()
+
+        return Validator(isValid,if(!isValid) "Invalid Add Bundled Transactions Request" else "")
+    }
```
Comment (ericbutz) [inline fiosdk/src/main/java/fiofoundation/io/fiosdk/FIOSDK.kt:4056]: Should we validate bundleSets, or just let the protocol handle error checking?

**Review by ericbutz — CHANGES_REQUESTED** [review]

File: fiosdk/src/main/java/fiofoundation/io/fiosdk/FIOSDK.kt (hunk 1 above)
Comment (edrotthoff) [reply fiosdk/src/main/java/fiofoundation/io/fiosdk/FIOSDK.kt]: will fix, thanks for seeing that..

File: fiosdk/src/main/java/fiofoundation/io/fiosdk/FIOSDK.kt (hunk 3 above)
Comment (edrotthoff) [reply fiosdk/src/main/java/fiofoundation/io/fiosdk/FIOSDK.kt:4056]: we check that the bundle sets is > 0. just to save a call to the protocol if its not appropriately set.

File: fiosdk/src/main/java/fiofoundation/io/fiosdk/FIOSDK.kt (hunk 2 above)
Comment (edrotthoff) [reply fiosdk/src/main/java/fiofoundation/io/fiosdk/FIOSDK.kt]: will rename. thanks for noticing this.

File: fiosdk/src/main/java/fiofoundation/io/fiosdk/FIOSDK.kt (hunk 3 above)
Comment (ericbutz) [reply fiosdk/src/main/java/fiofoundation/io/fiosdk/FIOSDK.kt:4056]: Sounds good.

**Review by ericbutz — APPROVED** [review]

**Pull request merged**